src/vs/workbench/contrib/notebook/browser/notebookServiceImpl.ts

```diff
@@ -48,12 +48,11 @@ export class NotebookProviderInfoStore extends Disposable {
 	private readonly _memento: Memento;
 	private _handled: boolean = false;
 
-	private _untitledCounter = 1;
-
 	private readonly _contributedEditors = new Map<string, NotebookProviderInfo>();
 	private readonly _contributedEditorDisposables = new DisposableStore();
 
 	constructor(
+		@INotebookService private readonly _notebookService: INotebookService,
```

Comment:
This is a cyclic dependency between the notebook service and this type. How about some indirection like a callback, e.g `existsCallback: (uri: URI) => boolean`?